datafusion/core/src/physical_planner.rs

```diff
@@ -958,10 +958,9 @@ impl DefaultPhysicalPlanner {
                                 })
                                 .collect::<Vec<_>>();
                             let projection =
-                                logical_plan::Projection::try_new_with_schema(
```

Comment:
here we shouldn't use `join_schema`, because `join_schema` may contain some column which don't exist in `project schema`